### PR TITLE
Async Database Client, Features, Bug Fixes

### PR DESCRIPTION
**Pull request by Scoder12**
Adds new Async Database Client, new utilities, and bug fixes from `dev` branch

### CODE REVIEW

**Comment by Codemonkey51** [comment]
Wait why do we have nest-asyncio if it is never used?

**Comment by Codemonkey51** [comment]
*as a dependancy

**Comment by Scoder12** [comment]
We don't?

**Comment by Codemonkey51** [comment]
These two images show the removal
2AED8F41-6D68-4FB4-8683-9100D6B95837
DACA94CC-403D-446F-B70A-9C02EC14B8A8

**Comment by Codemonkey51** [comment]
Into issue #10

File: src/replit/database/__init__.py [hunk 1]
```diff
@@ -295,58 +307,284 @@ def __repr__(self) -> str:
...
+        if self.dtype is not dict:
+            raise TypeError(
+                "Dictionary syntax can only be used if the datatype is dict"
+            )
+        data = self.get()
+        data[name] = value
+        self.set(data)
+
+
+class ReplitDb(dict):
```
Comment (kochman) [inline src/replit/database/__init__.py:486]: Why not inherit from AsyncReplitDb anymore?

Comment (Scoder12) [reply src/replit/database/__init__.py:486]: Whoops, that was a mistake! The sync code was restored from an old commit that didn't inherit

**Comment by Scoder12** [comment]
I accidentally added all of these commits to PR #12 so I'll use that instead

**Pull request closed without merging**